strax/context.py

```diff
@@ -904,6 +904,64 @@ def get_array(self, run_id: ty.Union[str, tuple, list],
             results = [x.data for x in source]
         return np.concatenate(results)
 
+    def get_accumulate(self,
+                       run_id: str,
+                       targets,
+                       function,
+                       fields,
+                       **kwargs):
+        """
+        Computes targets for run_id and accumulated result for specified
+        fields over all chunks.
+
+        Treats every not specified fields as a static parameter for
+        which the values of the very first entry of the very first
+        chunk are returned.
+
+        :param function: Function which shall be used to accumulate
+            the results. The function has to take two arguments. The
+            first one has to be the data which we read in, the second
+            argument must be the fields which should be accumulated. See
+            also the Example for more information.
+        :param fields: Column names of the data which should be
+            accumulated.
+
+        :returns dictionary: Dictionary with the keys of the specified
+            target and additional fields returned by the specified
+            function. Also added an additional field nchunks, which stores
+            the number of chunks.
+
+        Example:
+
+        """
+        nchunks = 0
+        for chunk in self.get_iter(run_id, targets, **kwargs):
+            data = chunk.data
+            chunk_start = chunk.start
+            chunk_end = chunk.end
+
+            if not nchunks:
+                # Initiate result and put common values:
+                res = {}
+                for name in data.dtype.names:
+                    if not name in fields:
+                        res[name] = data[0][name]  # Take static values from very first chunk.
+                res['time'] = chunk_start  # Easier to simply overwrite value.
+
+            # Now accumulate other values:
+            chunk_res = function(data, fields)
+            for key, value in chunk_res.items():
+                if not nchunks:
+                    res[key] = value
+                else:
+                    res[key] += value
```

Comment:
I think I implemented now your suggestions. Though, I did not see how I could implement the user specific function argument that one could compute e.g. the std or mean of a parameter over the entire run. I think here you still have to use get_iter.